### PR TITLE
Resetting Password 

### PR DESCRIPTION
**Pull request by rubencmd**
Resetting a password as well as confirming emails now works, still need to make redirect pages for the resetting password messages

### CODE REVIEW

File: src/main/java/ex/google/faculty_schedule_preference/user/token/ConfirmationTokenService.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    public void saveConfirmationToken(ConfirmationToken token){
+        confirmationTokenRepository.save(token);
+    }
+
+    public Optional<ConfirmationToken> getToken(String token){
+        return confirmationTokenRepository.findByToken(token);
+    }
+
+    public int setConfirmedAt(String token){
```
Comment (RyanAbukarn) [inline src/main/java/ex/google/faculty_schedule_preference/user/token/ConfirmationTokenService.java:28]: same this function is not needed it. Add a setter for confirmedAt

File: src/main/java/ex/google/faculty_schedule_preference/user/token/ConfirmationToken.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    @Column(nullable = false)
+    private String token;
+
+    @Column(nullable = false)
+    private LocalDateTime createdAt;
+
+    @Column(nullable = false)
+    private LocalDateTime expiresAt;
+
+    private LocalDateTime confirmedAt;
```
Comment (RyanAbukarn) [inline src/main/java/ex/google/faculty_schedule_preference/user/token/ConfirmationToken.java:39]: add a setter confirmedAt

File: src/main/java/ex/google/faculty_schedule_preference/user/UserRepository.java [hunk 3]
```diff
@@ -15,4 +16,9 @@
     boolean existsByEmail(String string);
 
     boolean existsByUsername(String username);
+
+    @Query("SELECT c FROM User c WHERE c.email = ?1")
```
Comment (RyanAbukarn) [inline src/main/java/ex/google/faculty_schedule_preference/user/UserRepository.java:20]: JPA understands to find by email so remove the @Query.

**Comment by rubencmd** [comment]
All feedback you gave has been updated. If you can give the changes a look over to make sure I implemented them correctly, that would be great.

**Comment by jozsef123** [comment]
added entitlement changes and email validation for login page

**Review by RyanAbukarn — APPROVED** [review]
looks good

**Pull request merged**